### DISCUSSION TITLE
[Question] How do I get debug logs?

### DISCUSSION THREAD

**Discussion by BallonLac** [2fast-team/2fast#111] [Q&A] [answered]
Hi there!  

I'm using 2fast and was very happy with it ... until I ran into some problems.  
2fast is freezing and stopping randomly, and often need 5-10 times restart before I can copy a code :/  

So, is there any way to have a debug mode enabled, a trace running, or something similar ?  
I run Windows 11, up to date of course.  

Thanks, and best regards.

**Comment by jp-weber** [comment]
Hello :)

There is currently no detailed log function that can be activated. However, after a crash there is currently a dialog with the system error message, which can be detailed or verbose.

Has such an error message appeared?

I have added this to the planned functions:
#112

**Reply by BallonLac** [reply]
Hi!

Thanks for #112 😃 .  

Unfortunately, no error message or dialog of any kind.  
Next time, I'll dig into Windows Event Log...

**Reply by BallonLac** [reply]
Hi,
Here is what I could find in Windows Application event log:

```
Faulting application name: Project2FA.UWP.exe, version: 1.0.0.0, time stamp: 0x65511426
Faulting module name: Windows.UI.Xaml.dll, version: 10.0.22621.3085, time stamp: 0x23511cf2
Exception code: 0xc0000005
Fault offset: 0x000000000010c9c0
Faulting process id: 0x0x4BF4
Faulting application start time: 0x0x1DA5B2D607DA8EB
Faulting application path: C:\Program Files\WindowsApps\38343JanPhilippWeber.2fastTwoFactorAuthenticatorSu_1.2.7.0_x64__nxr4mypqfqb9c\Project2FA.UWP.exe
Faulting module path: C:\Windows\System32\Windows.UI.Xaml.dll
Report Id: af40052a-4901-4966-a5dd-8ff782b25c25
Faulting package full name: 38343JanPhilippWeber.2fastTwoFactorAuthenticatorSu_1.2.7.0_x64__nxr4mypqfqb9c
Faulting package-relative application ID: App
``` 

There is also a WER file.

**Comment by BallonLac** [comment]
For your information, since last Windows update I can't reproduce the problem anymore :)  
Anyway, the debug log will be of help if any problem arise again!

**Comment by jp-weber** [comment, marked as answer]
The error log can be called up without logging into the app, or via the settings/functions

The command to call the log (if available):
https://github.com/2fast-team/2fast/wiki/Open-(error)-log-via-cmd

In future, the level under the settings will be made more detailed with ‘complete’, currently only the most important methods are logged.

**Comment by BallonLac** [comment]
Hi there,

2Fast is randomly hanging again, same error message in Windows Apps log as previously posted.

Nothing in log, despite Error logging set to "Complete". Version is 1.3.1.0.

**Comment by jp-weber** [comment]
Due to this behaviour, it is difficult to determine the cause, especially if the app does not even write a log. Other users had expressed similar strange behaviour after using third-party tools such as CCleaner.

**Comment by BallonLac** [comment]
I could never find a cause for my problems, and no log was available, ever. So I migrated to 2FAGuard, sorry guys :/
